sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,555 @@
+            )
+
+        names, estimators_ = zip(*self.estimators)
+        self._validate_names(names)
+
+        n_isnone = np.sum(
+            [est in (None, 'drop') for est in estimators_]
+        )
+        if n_isnone == len(self.estimators):
+            raise ValueError(
+                "All estimators are None or 'drop'. At least one is required "
+                "to be an estimator."
+            )
+
+        if isinstance(self.predict_method, str):
+            if self.predict_method != 'auto':
+                raise AttributeError(
+                    "When 'predict_method' is a string, it should be 'auto'. "
+                    " Got {} instead.".format(self.predict_method)
+                )
+            predict_method = [self.predict_method] * len(estimators_)
+        else:
+            if len(self.estimators) != len(self.predict_method):
+                raise AttributeError(
+                    "When 'predict_method' is a list, it should be the same "
+                    "length as the list of estimators. Provided {} methods "
+                    "for {} estimators."
+                    .format(len(self.predict_method), len(self.estimators))
+                )
+            predict_method = self.predict_method
+
+        self.predict_method_ = [
+            self._method_name(name, est, meth)
+            for name, est, meth in zip(names, estimators_, predict_method)
+        ]
+
+        # Fit the base estimators on the whole training data. Those
+        # base estimators will be used in transform, predict, and
+        # predict_proba. They are exposed publicly.
+        self.estimators_ = Parallel(n_jobs=self.n_jobs)(
+            delayed(_parallel_fit_estimator)(clone(est), X, y, sample_weight)
+            for est in estimators_ if est not in (None, 'drop')
+        )
+
+        # To train the meta-classifier using the most data as possible, we use
+        # a cross-validation to obtain the output of the stacked estimators.
+
+        # To ensure that the data provided to each estimator are the same, we
+        # need to set the random state of the cv if there is one and we need to
+        # take a copy.
+        random_state = check_random_state(self.random_state)
+        cv = check_cv(self.cv)
+        if hasattr(cv, 'random_state'):
+            cv.random_state = random_state
+
+        X_meta = Parallel(n_jobs=self.n_jobs)(
+            delayed(cross_val_predict)(clone(est), X, y, cv=deepcopy(cv),
+                                       method=meth, n_jobs=self.n_jobs,
+                                       verbose=self.verbose)
+            for est, meth in zip(estimators_, self.predict_method_)
+            if est not in (None, 'drop')
+        )
+
+        # Only not None or not 'drop' estimators will be used in transform.
+        # Remove the None from the method as well.
+        self.predict_method_ = [meth for meth in self.predict_method_
+                                if meth is not None]
```

Comment:
Is that equivalent to this? I was confused at first that `meth` could be None and thought the two lists (predict_method and estimators_) could be desynchronized. 


```python
        self.predict_method_ = [meth for (meth, est) in zip(self.predict_method_, estimators_)
                                if est not in (None, 'drop')]
```